src/messaging/MessageCounterSync.cpp

```diff
@@ -198,10 +198,10 @@ void MessageCounterSyncMgr::ProcessPendingGroupMsgs(NodeId peerNodeId)
         if (!entry.msgBuf.IsNull())
         {
             PacketHeader packetHeader;
-            uint16_t headerSize = 0;
 
-            if (packetHeader.Decode((entry.msgBuf)->Start(), (entry.msgBuf)->DataLength(), &headerSize) != CHIP_NO_ERROR)
```

Comment:
Looks like the tests don't actually check that this part is done right.  I tried undoing this change, and tests passed...  Followup is probably ok for adding a test here, but we should make sure we add it.

Reply:
To unit test this part, I need to make ProcessPendingGroupMsgs public, since our test framework only unit test public member function right now, but I don't think we should modify our code just for testing, any good idea?

Reply:
No good ideas yet.  Hence followup.